batcher/src/lib.rs

```diff
@@ -186,41 +184,59 @@ impl Batcher {
     ///     * Has the received block number surpassed the maximum interval with respect to the last posted batch block?
     /// An extra sanity check is made to check if the batch size is 0, since it does not make sense to post
     /// an empty batch, even if the block interval has been reached.
-    async fn batch_ready(&self, block_number: u64) -> bool {
-        let current_batch_size = self.current_batch.lock().await.len();
+    /// Once the batch is ready to be finalized, a copy of it is made to be passed to the `finalize_batch` function
+    /// and the current batch queue is cleared so that new data can be added to the batch while the old one is being
+    /// processed.
+    async fn is_batch_ready(&self, block_number: u64) -> Option<BatchQueue> {
```

Comment:
I would change the name of this method to something more descriptive

Reply:
+1